tests/providers/amazon/aws/hooks/test_base_aws.py

```diff
@@ -252,6 +322,78 @@ def import_mock(name, *args):
             [mock.call.get_default_id_token_credentials(target_audience='aws-federation.airflow.apache.org')]
         )
 
+    @unittest.skipIf(mock_sts is None, 'mock_sts package not present')
+    @mock.patch.object(AwsBaseHook, 'get_connection')
+    @mock_sts
+    def test_assume_role_with_saml(self, mock_get_connection):
```

Comment:
The title of this PR uses the word "Retry", but I don't see this word appearing in the tests. Does this code test the same functionality?